modules/spacecraftinstruments/shaders/renderableModel_fs.glsl

```diff
@@ -44,7 +46,13 @@ const float specularIntensity = 0.0;
 const float specularPower = 100.0;
 
 Fragment getFragment() {
-  vec4 textureColor = texture(baseTexture, vs_st);
+  vec4 textureColor;
+  if (has_texture_diffuse) {
+      textureColor = texture(baseTexture, vs_st);
+  }
+  else {
+      textureColor.rgb = baseColor;
```

Comment:
I'm not actually sure what the default value for a vec4 is, so maybe:  `textureColor = vec4(baseColor, 1.0);`   would be more legible?